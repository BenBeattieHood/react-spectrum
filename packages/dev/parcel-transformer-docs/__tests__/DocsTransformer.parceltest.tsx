import {join} from 'path';
import {MemoryFS} from '@parcel/fs';
import Parcel, {createWorkerFarm} from '@parcel/core';

const rootPath = join(__dirname, '..', '..', '..', '..');
const fixtureRoot = join(
  rootPath,
  'packages/dev/parcel-transformer-docs/__tests__/fixtures/parcel-transformer-test-app'
);

const workerFarm = createWorkerFarm();

afterAll(() => {
  workerFarm.end();
});

const outputFS = new MemoryFS(workerFarm);

const getParcelInstance = (workingDir: string) => {
  return new Parcel({
    config: join(rootPath, '.parcelrc'),
    entries: [join(workingDir)],
    targets: ['apiCheck'],
    outputFS,
    workerFarm,
  });
};

it('transforms assets with babel plugin', async () => {
  const parcel = getParcelInstance(fixtureRoot);
  await parcel.run();

  const code = JSON.parse(
    outputFS.readFileSync(join(fixtureRoot, 'dist', 'api.json'), 'utf8')
  );
  expect(code).toMatchSnapshot({
    exports: {
      App: {
        id: expect.stringMatching(/^.*\/packages\/.*:App/)
      },
      App2Real: {
        id: expect.stringMatching(/^.*\/packages\/.*:App2/)
      }
    }
  });
}, 50000);
