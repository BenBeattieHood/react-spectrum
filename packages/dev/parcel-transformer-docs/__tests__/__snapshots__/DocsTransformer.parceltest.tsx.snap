// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DocsTransformer - API writes export entry for React component 1`] = `
Object {
  "exports": Object {
    "App1": Object {
      "description": null,
      "id": StringMatching /\\^\\.\\*\\\\/test\\\\/src\\\\/index\\.tsx:App1/,
      "name": "App1",
      "props": null,
      "type": "component",
      "typeParameters": Array [],
    },
  },
  "links": Object {},
}
`;

exports[`DocsTransformer - API writes export entry for interface 1`] = `
Object {
  "exports": Object {
    "Foo": Object {
      "extends": Array [],
      "id": StringMatching /\\^\\.\\*\\\\/test\\\\/src\\\\/index\\.tsx:Foo/,
      "name": "Foo",
      "properties": Object {
        "a": Object {
          "default": null,
          "name": "a",
          "optional": false,
          "type": "property",
          "value": Object {
            "type": "number",
          },
        },
      },
      "type": "interface",
      "typeParameters": Array [],
    },
  },
  "links": Object {},
}
`;

exports[`DocsTransformer - API writes export entry for localName React component 1`] = `
Object {
  "exports": Object {
    "AppReal": Object {
      "description": null,
      "id": StringMatching /\\^\\.\\*\\\\/test\\\\/src\\\\/index\\.tsx:App2/,
      "name": "App2",
      "props": null,
      "type": "component",
      "typeParameters": Array [],
    },
  },
  "links": Object {},
}
`;

exports[`DocsTransformer - API writes export entry for referenced function 1`] = `
Object {
  "exports": Object {
    "d": Object {},
  },
  "links": Object {},
}
`;

exports[`DocsTransformer - API writes export entry for referenced string 1`] = `
Object {
  "exports": Object {
    "c": Object {},
  },
  "links": Object {},
}
`;

exports[`DocsTransformer - API writes export entry for static number 1`] = `
Object {
  "exports": Object {
    "a": Object {},
  },
  "links": Object {},
}
`;

exports[`DocsTransformer - API writes export entry for static string 1`] = `
Object {
  "exports": Object {
    "b": Object {},
  },
  "links": Object {},
}
`;

exports[`DocsTransformer - API writes export entry for type 1`] = `
Object {
  "exports": Object {
    "Foo": Object {
      "id": StringMatching /\\^\\.\\*\\\\/test\\\\/src\\\\/index\\.tsx:Foo/,
      "type": "link",
    },
  },
  "links": ObjectContaining {
    "/test/src/index.tsx:Foo": ObjectContaining {
      "id": StringMatching /\\^\\.\\*\\\\/test\\\\/src\\\\/index\\.tsx:Foo/,
    },
  },
}
`;

exports[`DocsTransformer - API writes export entry for type union 1`] = `
Object {
  "exports": Object {
    "Foo": Object {
      "id": StringMatching /\\^\\.\\*\\\\/test\\\\/src\\\\/index\\.tsx:Foo/,
      "type": "link",
    },
  },
  "links": ObjectContaining {
    "/test/src/index.tsx:Foo": ObjectContaining {
      "id": StringMatching /\\^\\.\\*\\\\/test\\\\/src\\\\/index\\.tsx:Foo/,
    },
  },
}
`;
